### ISSUE TITLE
Correct way to add edges to existing nodes (after commit)

### ISSUE THREAD

**Issue by schwab** [RedisGraph/redisgraph-py#16]
I'm trying to understand the proper way to find existing nodes in the graph and link them with edges.  What I've observed so far is that if I use 
`MERGE (x:thing {name:'uniqueid'})-[:points_at]->(y:otherthing {name:''uniqueid}) `
Then I end up with completely new and distinct x and y things with an edge between them, even if identical things already exist in the graph elsewhere. This is a problem especially if the original things have links to others and you later want to match things that have all the links.

It should be a way to add edges to the existing nodes and not have completely new ones created each time correct?

**Comment by schwab** [comment]
After some trial and error I found that this pattern works solves the issue quite well...

`MATCH (f:%s{%s:'%s'}), (t:%s{%s:'%s'}) CREATE (f)-[:in]->(t)`

**Comment by schwab** [comment]
One side effect of this approach is that it seems to use a lot more memory than expected.  For instance, while adding ~ 125k edges to a graph with ~50k nodes, redis cache total memory usage jumped  almost 600MB.  However, after performing redis save then restarting the redis service, the memory usage has dropped from ~ 750mb  to just 150MB.  Is there some kind of clean up step that needs to be performed after inserting a large number of edges?

**Comment by swilly22** [comment]
`MATCH (f:%s{%s:'%s'}), (t:%s{%s:'%s'}) CREATE (f)-[:in]->(t)`
Is the way to go about it, with regard to the increase in memory consumption it's something we'll need to look into.
Can you please provide the RedisGraph version you're using?
Run: `MODULE list` within redis-cli

**Comment by schwab** [comment]
Here's my graph version... ```4) 1) "name"
   2) "graph"
   3) "ver"
   4) (integer) 10004```  But I think I can safely witthdraw my claim that the MATCH CREATE calls are using more memory.  I've found 2 things that seemed to be contributing to that issue.  One, I was getting more than a single node for each item which should have been unique because data is added to the same node in more than one place in code and the 2nd->subsequent MATCH CREATE calls were creating duplicates.  After refactoring the code to first perform lookups based on some unique key before a create, this issue was corrected and the duplicates are not being made. It does mean however, that each CREATE now has to perform an extra call to check, but that's better than having dups.  The second issue was that, redis graph appears to hold a lot more memory than it actually needs.  After performing a lot of node and edges adds, if you then BGSAVE, Stop and Reload the redis service... the memory usually drops by 1/2 or more.  So, it appears we are perhaps using memory that's not released after the updates are done.  For now I just stop and restart after data import.  Is there a better way to free this temporary memory usage after inserts?
